shell/composables/useIsNewDetailPageEnabled.ts

```diff
@@ -1,14 +1,17 @@
 import { useRoute } from 'vue-router';
 import { LEGACY } from '@shell/config/query-params';
+import { computed } from 'vue';
 
 const enabledByDefault = true;
 
 export const useIsNewDetailPageEnabled = () => {
   const route = useRoute();
 
-  if (enabledByDefault) {
-    return route.query[LEGACY] !== 'true';
-  }
+  return computed(() => {
```

Comment:
It turns out that the value of useRoute can change asynchronously and can return undefined at the beginning depending on when it's called. I change this to a computed prop to react to those changes.